frontend/src/Components/ProgressBarForm.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect } from "react";
+import { LinearDeterminate } from "./ProgressBar";
+import Nav from "./Nav";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import "../CSS/ProgressBarForm.css";
+import { Link } from "react-router-dom";
+import axios from "axios";
+
+const ProgressBarForm = () => {
+  const [userId, setUserId] = useState(""); // Add userId state
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [isValid, setIsValid] = useState(false);
+
+  useEffect(() => {
+    const fetchUserProfile = async () => {
+      try {
+        if (userId) {
+          const response = await axios.get(
+            `http://localhost:3000/userProfile/${userId}`
+          );
+          const { firstName, lastName, id } = response.data;
+          setFirstName(firstName);
+          setLastName(lastName);
+          setUserId(id);
+          setIsValid(validateInput(firstName, lastName));
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchUserProfile();
+  }, [userId]);
```

Comment:
this is the dependency for the useEffect to work. Is userId changing ? I believe it should not, it should remain constant through out this page load. and when that is the case, we are NOT calling this function. So change the dependency or may be call the fetchUserProfile as a function whenever needed such as a button click.
My guess is there wont be ANY dependencies for this function to be called and set states for the user profile setup page